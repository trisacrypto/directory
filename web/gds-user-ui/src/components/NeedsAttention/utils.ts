export const enum AttentionAction {
  'VERIFY_EMAILS' = 'VERIFY_EMAILS',
  'VERIFY_PHONES' = 'VERIFY_PHONES',
  'VERIFY_ADDRESSES' = 'VERIFY_ADDRESSES',
  'VERIFY_IDENTITIES' = 'VERIFY_IDENTITIES',
  'VERIFY_DOCUMENTS' = 'VERIFY_DOCUMENTS',
  'VERIFY_ALL' = 'VERIFY_ALL',
  'VERIFY_NONE' = 'VERIFY_NONE'
}
