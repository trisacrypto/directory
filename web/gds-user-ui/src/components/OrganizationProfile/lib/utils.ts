export const StatusCheck = (status: string) => {
  switch (status) {
    case 'NO_VERIFICATION':
      return 'Not Verified';
    case 'PENDING_REVIEW':
      return 'Pending Review';
    case 'VERIFIED':
      return 'Verified';
    case 'REJECTED':
      return 'Rejected';
    default:
      return 'Pending Registration';
  }
};
