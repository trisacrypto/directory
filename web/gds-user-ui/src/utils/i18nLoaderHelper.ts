import { i18n } from '@lingui/core';
import { en, fr, de, zh, ja, ru } from 'make-plural/plurals';
import { Locales } from 'types/type';

export const DEFAULT_LOCALE: Locales = 'en';

export async function dynamicActivate(locale: string) {
  const { messages } = await import(`../locales/${locale}/messages`);

  i18n.load(locale, messages);
  i18n.loadLocaleData({
    en: { plurals: en },
    de: { plurals: de },
    fr: { plurals: fr },
    zh: { plurals: zh },
    ja: { plurals: ja }
  });
  i18n.activate(locale);
}
