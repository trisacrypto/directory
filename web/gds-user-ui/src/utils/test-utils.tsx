import { ChakraProvider } from '@chakra-ui/react';
import { render as rtlRender, RenderOptions as RTLRenderOptions } from '@testing-library/react';
import React from 'react';
import { FormProvider, useForm } from 'react-hook-form';
import { Provider } from 'react-redux';
import customTheme from './theme';
import rootReducer from 'application/store/rootReducer';
import { configureStore, Store } from '@reduxjs/toolkit';

type RenderOptions = {
  rhfDefaultValues?: { [x: string]: any };
  store?: Store;
  preloadedState?: StoreState;
  renderOptions?: Omit<RTLRenderOptions, 'wrapper'>;
};

type StoreState = {
  [x: string]: any;
};

export function render(
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  {
    rhfDefaultValues,
    preloadedState,
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  }: RenderOptions = {}
) {
  function Wrapper({ children }: { children: React.ReactNode }) {
    const methods = useForm({ defaultValues: rhfDefaultValues });
    return (
      <Provider store={store}>
        <ChakraProvider theme={customTheme}>
          <FormProvider {...methods}>{children}</FormProvider>
        </ChakraProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
