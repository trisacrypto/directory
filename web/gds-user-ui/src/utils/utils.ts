import { left } from '@popperjs/core';
import _ from 'lodash';
import { RegistrationAuthority, StepStatus } from 'types/type';
import { TStep } from './localStorageHelper';
import registrationAuthority from './registration-authority.json';

const DEFAULT_REGISTRATION_AUTHORITY = 'RA777777';

export const findStepKey = (steps: any, key: number) =>
  steps?.filter((step: any) => step.key === key);

export const isValidUuid = (str: string) => {
  // Regular expression to check if string is a valid UUID
  const regexExp =
    /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/gi;
  return regexExp.test(str);
};

export const getStepData = (steps: any, key: number): TStep | undefined => {
  const s = findStepKey(steps, key);
  if (s && s?.length === 1) {
    return s[0].data;
  }
  return undefined;
};

export const getStepStatus = (steps: any, key: number): StepStatus | undefined => {
  const s = findStepKey(steps, key);
  if (s && s?.length === 1) {
    return s[0].status;
  }
  return undefined;
};

export const hasStepError = (steps: any): boolean => {
  const s = steps.filter((step: any) => step.status === 'error');
  return s.length > 0;
};

export const getStepDatas = (steps: any) => {
  const s = steps
    ?.map((step: any) => step.data)
    .reduce((acc: any, cur: any) => ({ ...acc, ...cur }), {});

  return { ...s };
};

export const getValueByPathname = (obj: Record<string, any>, path: string) => {
  return _.get(obj, path);
};

export const getDomain = (url: string | URL) => {
  try {
    const _url = new URL(url);
    return _url?.hostname?.replace('www.', '');
  } catch (error) {
    console.error('[error]', error);
    return null;
  }
};

export const getRegistrationAuthorities = () => [...new Set(registrationAuthority)];

export const getRegistrationAuthoritiesOptions = (country?: any) => {
  const newArray = [...Array.from(new Set(registrationAuthority))];
  if (country) {
    return newArray
      .filter(
        (v: RegistrationAuthority) =>
          v.country === country || v.option === DEFAULT_REGISTRATION_AUTHORITY
      )
      .map((v: RegistrationAuthority) => {
        const label = v.organization ? `${v.option} - ${v.organization}` : `${v.option}`;
        const l =
          v.jurisdiction && v.jurisdiction !== v.country_name
            ? `${label} - ${v.jurisdiction}`
            : label;
        return {
          value: v.option,
          label: l,
          isDisabled: !!v.isDisabled
        };
      });
  }
  return newArray.map((v: RegistrationAuthority) => {
    const label = v.organization ? `${v.option} - ${v.organization}` : `${v.option}`;
    return {
      value: v.option,
      label,
      isDisabled: !!v.isDisabled
    };
  });
};

export const mapTrixoFormForBff = (data: any) => {
  const { trixo } = data;
  const { applicable_regulations, other_jurisdictions } = trixo;

  const cleanAppRegulation = applicable_regulations.reduce((acc: any, value: any) => {
    if (value.name.length > 0) {
      acc.push(value.name);
    }
    return acc;
  }, []);
  const cleanOtherJurisdiction = other_jurisdictions.filter(
    (o: any) => o?.country?.length > 0 && o?.regulator_name?.length > 0
  );

  return {
    ...data,
    trixo: {
      ...trixo,
      applicable_regulations: cleanAppRegulation.length > 0 ? cleanAppRegulation : [],
      other_jurisdictions: cleanOtherJurisdiction.length > 0 ? cleanOtherJurisdiction : []
    }
  };
};

export const hasValue = (obj: Record<string, any>) => {
  return obj && Object.values(obj).some(Boolean);
};

export const getColorScheme = (status: string) => {
  if (status === 'yes' || status) {
    return 'cyan';
  } else {
    return '#eee';
  }
};

export function currencyFormatter(
  amount: number | bigint,
  { style = 'currency', currency = 'USD' }
) {
  const formatedAmount = new Intl.NumberFormat('en-US', {
    style,
    currency
  });

  return formatedAmount.format(amount);
}
