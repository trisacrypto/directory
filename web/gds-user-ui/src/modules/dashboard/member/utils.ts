import { DirectoryType, VaspDirectoryEnum, DirectoryTypeEnum } from './memberType';
import { convertToCVS, downloadCSV } from 'utils/utils';
import { t } from '@lingui/macro';

export const memberTableHeader = [
  {
    key: 'name',
    label: t`Member Name`
  },
  {
    key: 'first_listed',
    label: t`Joined`
  },
  {
    key: 'last_updated',
    label: t`Last Updated`
  },
  {
    key: 'registered_directory',
    label: t`Network`
  },
  {
    key: 'status',
    label: t`Status`
  }
];
export const getVaspDirectory = (dir: DirectoryType) => {
  return dir === DirectoryTypeEnum.TESTNET ? VaspDirectoryEnum.TESTNET : VaspDirectoryEnum.MAINNET;
};

export const getVaspNetwork = (dir: any) => {
  switch (dir) {
    case (VaspDirectoryEnum.TESTNET, VaspDirectoryEnum.TESTNET_DEV):
      return 'TestNet';
    case (VaspDirectoryEnum.MAINNET, VaspDirectoryEnum.MAINNET_DEV):
      return 'MainNet';
    default:
      return 'MainNet';
  }
};

export const getVaspStatus = (status: number) => {
  switch (status) {
    case 1:
      return t`NO VERIFICATION`;
    case 2:
      return t`SUBMITTED`;
    case 3:
      return t`EMAIL VERIFIED`;
    case 4:
      return t`PENDING REVIEW`;
    case 5:
      return t`REVIEWED`;
    case 6:
      return t`VERIFIED`;
    case 7:
      return t`REJECTED`;
    case 8:
      return t`APPEALED`;
    case 9:
      return t`ERRORED`;
    default:
      return t`NO VERIFICATION`;
  }
};

export const downloadMembers2CVS = (member: any) => {
  const m = member.map((item: any) => {
    return {
      ...item,
      status: getVaspStatus(item.status),
      registered_directory: getVaspNetwork(item.registered_directory)
    };
  });
  const memberCsv = convertToCVS(m, memberTableHeader as ITableHeader[]);
  downloadCSV(memberCsv, 'members');
};

