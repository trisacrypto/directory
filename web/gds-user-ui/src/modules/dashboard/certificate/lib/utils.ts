import { basicDetailsValidationSchema } from 'modules/dashboard/certificate/lib/basicDetailsValidationSchema';
import { legalPersonValidationSchemam } from 'modules/dashboard/certificate/lib/legalPersonValidationSchema';
import { contactsValidationSchema } from 'modules/dashboard/certificate/lib/contactsValidationSchema';
import { trisaImplementationValidationSchema } from 'modules/dashboard/certificate/lib/trisaImplementationValidationSchema';
import { trixoQuestionnaireValidationSchema } from 'modules/dashboard/certificate/lib/trixoQuestionnaireValidationSchema';

export const isBasicDetailsCompleted = async (data: any) => {
  try {
    const r = await basicDetailsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    return false;
  }
};

export const isLegalPersonCompleted = async (data: any) => {
  try {
    const r = await legalPersonValidationSchemam.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('legal person validation failed', err);
    return false;
  }
};

export const isContactsCompleted = async (data: any) => {
  try {
    const r = await contactsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('contacts validation failed', err);
    return false;
  }
};

export const isTrisaImplementationCompleted = async (data: any) => {
  try {
    await trisaImplementationValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};

export const isTrixoQuestionnaireCompleted = async (data: any) => {
  try {
    await trixoQuestionnaireValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};
