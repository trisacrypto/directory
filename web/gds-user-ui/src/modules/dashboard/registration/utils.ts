import { getDefaultValue } from 'components/BasicDetailsForm/validation';
import { getRegistrationDefaultValues } from 'modules/dashboard/certificate/lib';
import {
  postRegistrationData,
  getRegistrationData,
  getSubmissionStatus
} from 'modules/dashboard/registration/service';
import { handleError } from 'utils/utils';

export const postRegistrationValue = (data: any) => {
  console.log('[postRegistrationValue]', data);
  return new Promise((resolve, reject) => {
    postRegistrationData(data)
      .then((res) => {
        console.log('[postRegistrationData]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        console.log('[postRegistrationData]', err);
        reject(err);
      });
  });
};
export const getRegistrationDefaultValue = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return regData.data;
    }
    const defaultValue: any = localStorage.getItem('certificateForm');
    if (defaultValue) {
      const val = JSON.parse(defaultValue);
      const postData = await postRegistrationData(val);
      if (postData.status === 204) {
        const getData = await getRegistrationData();
        localStorage.removeItem('certificateForm');
        return getData.data;
      }
    }
    const v = getRegistrationDefaultValues();
    await postRegistrationValue(v);
    return getRegistrationDefaultValues();
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

export const setRegistrationDefaultValue = () => {
  console.log('[setRegistrationDefaultValue]');
  const defaultValue: any = getRegistrationDefaultValues();
  return new Promise((resolve, reject) => {
    postRegistrationData(defaultValue)
      .then((res) => {
        // console.log('[default postRegistration value]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        handleError(err, 'failed to post registration value');
        reject(err);
      });
  });
};

// get registration data from backend and download the file
export const downloadRegistrationData = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      const blob = new Blob([JSON.stringify(regData.data)], { type: 'application/json' });
      const url = window.URL.createObjectURL(blob);
      const link = document.createElement('a');
      link.href = url;
      link.setAttribute('download', 'registration.json');
      document.body.appendChild(link);
      link.click();
    }
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

// load default stepper

export const getDefaultStepper = async () => {
  try {
    const [regData, regStatus] = await Promise.all([getRegistrationData(), getSubmissionStatus()]);
    console.log('[regStatus]', regStatus.data);
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return {
        currentStep: regData.data.state.current,
        steps: regData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: regData.data.state.ready_to_submit,
        testnetSubmitted: regStatus?.data?.testnetSubmitted || false,
        mainnetSubmitted: regStatus?.data?.mainnetSubmitted || false
      };
    }
    const defaultValue: any = {
      current: 1,
      steps: [
        {
          key: 1,
          status: 'progress'
        }
      ]
    };
    // update registrations state object
    const postData = await postRegistrationData({ state: defaultValue });
    if (postData.status === 204) {
      const getData = await getRegistrationData();
      return {
        currentStep: getData.data.state.current,
        steps: getData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: false,
        testnetSubmitted: false,
        mainnetSubmitted: false
      };
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};

// // load default stepper without async call
// export const loadDefaultStepperSync = () => {
//   return Promise.resolve({
//     currentStep: 1,
//     steps: [
//       {
//         key: 1,
//         status: 'progress'
//       }
//     ],
//     lastStep: null,
//     hasReachSubmitStep: false
//   });
// };

// set stepper data
export const getRegistrationAndStepperData = async () => {
  try {
    const [regData, regStatus] = await Promise.all([
      getRegistrationDefaultValue(),
      getSubmissionStatus()
    ]);
    if (regData) {
      const response: any = {
        registrationData: regData,
        stepperData: {
          currentStep: regData?.state?.current || 1,
          steps: regData?.state?.steps || [{ key: 1, status: 'progress' }],
          lastStep: null,
          hasReachSubmitStep: regData?.state?.ready_to_submit || false,
          testnetSubmitted: !!regStatus?.data?.testnet_submitted,
          mainnetSubmitted: !!regStatus?.data?.mainnet_submitted
        }
      };
      return response;
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};
