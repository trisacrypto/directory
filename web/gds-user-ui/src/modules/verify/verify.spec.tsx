import { dynamicActivate } from 'utils/i18nLoaderHelper';
import { render } from 'utils/test-utils';
import VerifyPage from '.';

describe('<VerifyPage />', () => {
  beforeEach(() => {
    dynamicActivate('en');
  });

  it('should match snapshot', () => {
    const { container } = render(<VerifyPage />);
    expect(container).toMatchSnapshot();
  });
});
