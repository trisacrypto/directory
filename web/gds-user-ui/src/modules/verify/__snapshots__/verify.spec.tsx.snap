// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerifyPage /> should match snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-1jtm99b"
  >
    <div
      class="css-1pwxvtk"
    >
      <div
        class="chakra-container css-1a3ltpp"
      >
        <div
          class="css-104xflu"
        >
          <div
            class="css-1uuxm7r"
          >
            <div
              class="css-0"
            >
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                <a
                  class="chakra-link css-htwawm"
                >
                  <div
                    class="css-1p293e5"
                  >
                    <img
                      alt="Trisa logo"
                      src="trisa_logo.svg"
                    />
                  </div>
                </a>
              </a>
            </div>
            <div
              class="css-1ydjxjn"
            >
              <svg
                viewBox="0 0 20 20"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Menu
                </title>
                <path
                  d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
                  fill="black"
                />
              </svg>
            </div>
            <div
              class="chakra-stack css-sz9ruu"
            >
              <div
                class="chakra-stack css-1p6wuvz"
              >
                <div
                  class="chakra-select__wrapper css-aq64ov"
                >
                  <select
                    class="chakra-select css-k7r2wc"
                  >
                    <option
                      value="en"
                    >
                      ðŸ‡ºðŸ‡¸ EN
                    </option>
                    <option
                      value="fr"
                    >
                      ðŸ‡«ðŸ‡· FR
                    </option>
                    <option
                      value="de"
                    >
                      ðŸ‡©ðŸ‡ª DE
                    </option>
                    <option
                      value="zh"
                    >
                      ðŸ‡¨ðŸ‡³ ZH
                    </option>
                    <option
                      value="ja"
                    >
                      ðŸ‡¯ðŸ‡µ JA
                    </option>
                  </select>
                  <div
                    class="chakra-select__icon-wrapper css-zj2dqk"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-select__icon"
                      focusable="false"
                      role="presentation"
                      style="width: 1em; height: 1em; color: currentColor;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="chakra-text css-dme2i4"
              >
                <a
                  href="/#about"
                >
                  <a
                    class="chakra-link css-htwawm"
                  >
                    About TRISA
                  </a>
                </a>
              </div>
              <div
                class="chakra-text css-dme2i4"
                data-testid="documentation"
              >
                <a
                  class="chakra-link css-htwawm"
                  href="https://trisa.dev/en"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Documentation
                </a>
              </div>
              <div
                class="chakra-stack css-n21gh5"
              >
                <a
                  class=""
                  href="/auth/login"
                >
                  <button
                    class="chakra-button css-1ncsb74"
                    type="button"
                  >
                    Login
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-i9gxme"
    >
      <div
        class="css-at8wy6"
      >
        <div
          class="chakra-stack css-156zt38"
        >
          <div
            class="chakra-alert css-x6jk6h"
            role="alert"
          >
            <span
              class="chakra-alert__icon css-123gu50"
            >
              <svg
                class="chakra-icon css-fi0ww0"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <div
              class="chakra-alert__title css-1gvuxlj"
            >
              Unable to Verify Contact
            </div>
            <div
              class="chakra-alert__desc css-1rqsjo7"
            >
              Invalid params
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      style="width: 100%;"
    >
      <div
        class="chakra-stack css-1arqzha"
      >
        <div
          class="css-248241"
        >
          <p
            class="chakra-text css-q7d5fr"
          >
            A component of
             
            <a
              class="chakra-link css-1av0s05"
              href="https://trisa.io"
            >
              the TRISA architecture
            </a>
             
            for Cryptocurrency Travel Rule compliance.
          </p>
          <p
            class="chakra-text css-q7d5fr"
          >
            Created and maintained by
             
            <a
              class="chakra-link css-1av0s05"
              href="https://rotational.io"
            >
               
              Rotational Labs
            </a>
             
            in partnership with
             
            <a
              class="chakra-link css-1av0s05"
              href="https://cyphertrace.com"
            >
               
              CipherTrace
            </a>
             
            on behalf of
             
            <a
              class="chakra-link css-1av0s05"
              href="https://trisa.io"
            >
              TRISA
            </a>
             
          </p>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
